### PR TITLE
AmountSettingsHandler: floor --clboss-min-onchain at CLN's min-emergency floor (30000 sat)

### PR DESCRIPTION
**Pull request by Amperstrand**
## Problem

`--clboss-min-onchain` is accepted at any value ≥ 0.  CLN itself
refuses to let the wallet spend below `min-emergency-msat`
(25 000 sat by default) plus funding fees — so any CLBOSS reserve
below that floor produces a **permanent channel-creation failure
loop**: CLBOSS computes `worth = onchain − reserve`, plans
channel(s) spending the wallet down past CLN's floor, the carpenter's
`multifundchannel` fails with error 313 ("We would not have enough
left for min-emergency-msat 25000sat"), the attempted candidates are
removed from the candidate table, the next cycle re-proposes them —
forever.  Observed live on regtest: a node with
`--clboss-min-onchain=10000` loops 313 with zero channels ever
created; the same node with the floor applied lands both planned
channels.

The trap is wealth-level-dependent (an over-funded wallet never
spends down to the reserve, which hides it in testing) and
silent (nothing validates the config at startup).

## Fix

`AmountSettingsHandler` End-of-options validation now coerces the
reserve below `min_usable_reserve = 30000` sat (= CLN's 25 000 sat
default floor + funding-fee headroom) to that floor, with a `Warn`
naming the mechanism.  The floor equals the existing default, so
default configurations are no-ops.

## Tests

Table-driven cases `10000 → 30000` and `29999 → 30000` added to
`tests/boss/test_amountsettingshandler.cpp` (plus the existing
default no-op cases).  `make check` green.

Live validation (regtest A/B, identical flags, funded
back-to-back, deciders firing on the same block with the same fee
verdict): unpatched loops `multifundchannel` 313 with 0 channels;
patched logs the Warn and lands both channels CHANNELD_NORMAL.

- Enforce a minimum `--clboss-min-onchain` reserve of 30,000 sat during end-of-options validation.
- Log a warning when a value below the minimum is raised to 30,000 sat. Keep the existing default unchanged.
- Add table-driven tests for reserves below, at, and above the minimum.

### CODE REVIEW

**Review by ksedgwic — CHANGES_REQUESTED** [review]
Thanks, this is a real trap and the fix is the right shape: coerce
with a warning, the same way the handler already treats
clboss-min-channel and clboss-max-channel.  The three amount options
are not registered dynamic, so the check at EndOfOptions is the only
place it is needed.  A few requests before merging:

1. Commit title and test comment: drop the internal "F-3" label.
   Something like "AmountSettingsHandler: floor clboss-min-onchain
   at CLN's emergency reserve" for the title, and the test comment
   without the tag.

2. README: one sentence under --clboss-min-onchain saying the value
   is floored at 30000 sat, which is CLN's default min-emergency-msat
   plus room for the funding fee, and that an operator who raises
   min-emergency-msat should raise clboss-min-onchain by the same
   amount.

3. CHANGELOG: a bullet under "### Fixed" in a new "## [0.17.1] -
   Unreleased" section at the top (several open PRs add that
   header; I fold them on merge).

4. The Warn text: shorter, and say what it assumes.  For example:
   "clboss-min-onchain %u is below 30000 sat, CLN's default
   min-emergency-msat plus funding fees; every channel open would
   fail.  Using 30000."

5. CodeRabbit's minor finding holds: parse_sats reads into an
   unsigned, so "-1" becomes a huge value and bypasses the floor.  A
   digits-only check in parse_sats with a "-1" test case would close
   it.  Small enough to do here; say so if you would rather keep it
   separate.

6. CodeRabbit's other point, following the effective
   min-emergency-msat instead of a constant: not for this PR.  The
   Initiator already fetches listconfigs at init, so a follow-up
   could take the floor from there.  The README sentence in item 2
   covers the gap until then.

**Comment by ksedgwic** [comment]
@Amperstrand ping

**Review by ksedgwic — CHANGES_REQUESTED** [review]
Thanks, items 1 to 5 are all in. One thing before merging, from
CodeRabbit's finding, which holds: the digits-only check turns any
spelling that is not bare digits into 0, so a value that parsed
before, such as "500000sat" or "+500000", now floors to 30000 and
lowers the reserve. The wrap bug is only the leading minus, so
reject that alone and keep the stream parse for everything else:

    if (str.empty() || str[0] == '-')
        return Ln::Amount::sat(0);
    auto is = std::istringstream(str);
    is >> sats;

with the comment adjusted, and one table row showing a value with
trailing text still parses as before. Then this is ready; it goes
into the 0.17.1 integration branch test ...

File: Boss/Mod/AmountSettingsHandler.cpp [hunk 1]
```diff
@@ -40,6 +40,15 @@ auto const additional_remaining = Ln::Amount::sat(20000);
...
 	(1.0 + trigger_factor) * min_min_channel + additional_remaining;
 
+/* CLBOSS computes the channel-creation budget as
+ *   onchain_spendable - reserve
+ * and spends down to that level, but CLN refuses any funding
+ * transaction that would leave less than its `min-emergency-msat`
+ * (25,000 sat default) plus the funding fee.  A reserve below
+ * that floor makes every multifundchannel fail with error 313
+ * forever while churning the candidate table, so floor it.  */
+auto const min_usable_reserve = Ln::Amount::sat(30000);
```
[the remark this answers is not in the document]
Comment (ksedgwic) [reply Boss/Mod/AmountSettingsHandler.cpp]: Not for this PR. The floor stays a constant, and the README tells an
operator who raises min-emergency-msat to raise clboss-min-onchain by
the same amount. Reading it from listconfigs is a follow-up; the
Initiator already fetches listconfigs at init.

[the remark this answers is not in the document]
Comment (ksedgwic) [reply Boss/Mod/AmountSettingsHandler.cpp]: Not for this PR. The floor stays a constant, and the README tells an
operator who raises min-emergency-msat to raise clboss-min-onchain by
the same amount. Reading it from listconfigs is a follow-up; the
Initiator already fetches listconfigs at init.
